presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -951,6 +951,16 @@ public void testParseDuration()
         assertInvalidFunction("parse_duration('abc')", "duration is not a valid data duration string: abc");
     }
 
+    @Test
+    public void testLastDayOfMonth()
+    {
+        DateTime result = DATE.withDayOfMonth(31);
```

Comment:
Made changes for this, can you please take a look at https://github.com/prestodb/presto/pull/8545 request. I opened duplicate pull request by mistake but it has latest review comments so keeping it open and closing this